### PR TITLE
Refine the spill threshold for aggregation/sort, refine the memory usage calculation for aggregation

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: ref #6528

Problem Summary:

### What is changed and how it works?
1. refine the spill threshold for aggregation/sort, now if aggregation/sort is executed parallelly, the threshold used in every sub agg/sort is recalculated by total threshold and the concurrency
2. refine the memory usage calculation for aggregation, don't rely on the task-level memory tracker to calculate the memory usage of aggregation.
### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Interpreters/Aggregator.h [hunk 1]
```diff
@@ -718,7 +718,7 @@ struct AggregatedDataVariants : private boost::noncopyable
     case Type::NAME:                                                                                       \
     {                                                                                                      \
         const auto * ptr = reinterpret_cast<const AggregationMethodName(NAME) *>(aggregation_method_impl); \
-        return ptr->data.size() + (without_key != nullptr);                                                \
+        return ptr->data.size();                                                                           \
```
Comment (yibin87) [inline dbms/src/Interpreters/Aggregator.h:721]: why remove the (without_key != nullptr) check here?

Comment (windtalker) [reply dbms/src/Interpreters/Aggregator.h:721]: It is used by `over_flow_row`, and we already remove the support of `over_flow_row`

**Review by yibin87 — APPROVED** [review]
LGTM

File: dbms/src/Flash/Coprocessor/InterpreterUtils.cpp [hunk 2]
```diff
@@ -120,7 +120,7 @@ void orderStreams(
                 order_descr,
                 settings.max_block_size,
                 limit,
-                settings.max_bytes_before_external_sort,
+                getAverageThreshold(settings.max_bytes_before_external_sort, std::min(max_streams, pipeline.streams.size())),
```
Comment (SeaRise) [inline dbms/src/Flash/Coprocessor/InterpreterUtils.cpp:123]: I prefer
```
auto avg_threshold = getAverageThreshold(settings.max_bytes_before_external_sort, pipeline.streams.size());
stream = std::make_shared<MergeSortingBlockInputStream>(xxxx, avg_threshold);
```
The count of sort stream is `pipeline.streams.size()` at this time, not `min(max_streams, pipeline.streams.size())`, even if `max_streams < pipeline.streams.size()`. So we shoud use `pipeline.streams.size()` or not `std::min(max_streams, pipeline.streams.size())` here?

Comment (windtalker) [reply dbms/src/Flash/Coprocessor/InterpreterUtils.cpp]: Done

**Review by SeaRise — APPROVED** [review]
lgtm

**Comment by windtalker** [comment]
hold after code freeze

**Pull request merged**